src/utils/pickupTimeEstimator.ts

```diff
@@ -117,30 +122,58 @@ export const getPickupTimeEstimate = async (
   )
   rationale.push({
     time: destinationServiceTime,
-    activity: `destination (${deliveryLocationId}) service time`,
+    activity: `Destination (${deliveryLocationId}) service time given ${
+      formatDateAndTime(new Date(arrivalAtDestination)).time
+    } arrival`,
   })
 
-  // Account for travel to reading room:
-  let arrivalAtHoldshelf = await addOnsiteTravelDuration(
-    destinationServiceTime,
-    deliveryLocationId
+  // Adjust to special delivery schedules for special rooms:
+  const { hasSpecialDeliverySchedule } = adjustToSpecialSchedule(
+    deliveryLocationId,
+    destinationServiceTime
   )
-  rationale.push({ time: arrivalAtHoldshelf, activity: "onsite travel time" })
 
-  // Adjust to special delivery schedules for special rooms:
-  const hasSpecialDelivery = hasSpecialDeliverySchedule(deliveryLocationId)
-  if (hasSpecialDelivery) {
-    arrivalAtHoldshelf = adjustToSpecialSchedule(
-      deliveryLocationId,
-      arrivalAtHoldshelf
+  // If delivered to a scholar room and requested at least 30mins before opening, skip travel time:
+  const openingHours = await getNextServiceHours(
+    deliveryLocationId,
+    destinationServiceTime
+  )
+  const skipOnsiteTravelTime =
+    hasSpecialDeliverySchedule &&
+    timestampIsLTE(
+      arrivalAtDestination,
+      addMinutes(openingHours.startTime, -30)
+    )
+
+  let arrivalAtHoldshelf = destinationServiceTime
+
+  // Unless deliverying to a scholar room before opening hours, on-site apply travel time:
+  if (!skipOnsiteTravelTime) {
+    arrivalAtHoldshelf = await addOnsiteTravelDuration(
+      destinationServiceTime,
+      deliveryLocationId
     )
+    rationale.push({ time: arrivalAtHoldshelf, activity: "Onsite travel time" })
+  }
+
+  if (hasSpecialDeliverySchedule) {
+    let specialDeliveryRationale
+    ;({ arrivalAtHoldshelf, rationale: specialDeliveryRationale } =
```

Comment:
You shouldn't need the semicolon and parens here.

Reply:
I get "Declaration or statement expected. This '=' follows a bloc...ent, you might need to wrap the whole assignment in parentheses" without them